cmd/prysmctl/p2p/client.go

```diff
@@ -0,0 +1,230 @@
+}
+
+func (c *client) MetadataSeq() uint64 {
+	return c.meta.SequenceNumber()
+}
+
+// Send a request to specific peer. The returned stream may be used for reading,
+// but has been closed for writing.
+// When done, the caller must Close() or Reset() on the stream.
+func (c *client) Send(
+	ctx context.Context,
+	message interface{},
+	baseTopic string,
+	pid peer.ID,
+) (corenet.Stream, error) {
+	ctx, span := trace.StartSpan(ctx, "p2p.Send")
+	defer span.End()
+	topic := baseTopic + c.Encoding().ProtocolSuffix()
+	span.AddAttributes(trace.StringAttribute("topic", topic))
+
+	// Apply max dial timeout when opening a new stream.
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+
+	stream, err := c.host.NewStream(ctx, pid, protocol.ID(topic))
+	if err != nil {
+		tracing.AnnotateError(span, err)
+		return nil, errors.Wrap(err, "could not open new stream")
+	}
+	// do not encode anything if we are sending a metadata request
+	if baseTopic != p2p.RPCMetaDataTopicV1 && baseTopic != p2p.RPCMetaDataTopicV2 {
+		castedMsg, ok := message.(ssz.Marshaler)
+		if !ok {
+			return nil, errors.Errorf("%T does not support the ssz marshaller interface", message)
+		}
+		if _, err := c.Encoding().EncodeWithMaxLength(stream, castedMsg); err != nil {
+			tracing.AnnotateError(span, err)
+			_err := stream.Reset()
+			_ = _err
+			return nil, err
+		}
+	}
+	// Close stream for writing.
+	if err := stream.CloseWrite(); err != nil {
+		tracing.AnnotateError(span, err)
+		_err := stream.Reset()
+		_ = _err
+		return nil, errors.Wrap(err, "could not close write")
+	}
+
+	return stream, nil
+}
+
+func (c *client) retrievePeerAddressesViaRPC(ctx context.Context, beaconEndpoints []string) ([]string, error) {
+	if len(beaconEndpoints) == 0 {
+		return nil, errors.New("no beacon RPC endpoints specified")
+	}
+	peers := make([]string, 0)
+	for i := 0; i < len(beaconEndpoints); i++ {
+		conn, err := grpc.Dial(beaconEndpoints[i], grpc.WithInsecure())
+		if err != nil {
+			return nil, err
+		}
+		nodeClient := pb.NewNodeClient(conn)
+		hostData, err := nodeClient.GetHost(ctx, &emptypb.Empty{})
+		if err != nil {
+			return nil, err
+		}
+		if len(hostData.Addresses) == 0 {
+			continue
+		}
+		peers = append(peers, hostData.Addresses[0]+"/p2p/"+hostData.PeerId)
+	}
+	return peers, nil
+}
+
+func (c *client) initializeMockChainService(ctx context.Context) (*mockChain, error) {
+	genesisResp, err := c.nodeClient.GetGenesis(ctx, &emptypb.Empty{})
+	if err != nil {
+		return nil, err
+	}
+	currEpoch := slots.ToEpoch(slots.SinceGenesis(genesisResp.GenesisTime.AsTime()))
+	currFork, err := forks.Fork(currEpoch)
+	if err != nil {
+		return nil, err
+	}
+	return &mockChain{
+		genesisTime:     genesisResp.GenesisTime.AsTime(),
+		currentFork:     currFork,
+		genesisValsRoot: bytesutil.ToBytes32(genesisResp.GenesisValidatorsRoot),
+	}, nil
+}
+
+// Retrieves an external ipv4 address and converts into a libp2p formatted value.
+func ipAddr() net.IP {
```

Comment:
A lot of these methods are direct copies of the p2p methods. It might be better to move them to a common sub-package in `network` . Same feedback as the other methods